actioncable/lib/action_cable/server/base.rb

```diff
@@ -41,6 +41,9 @@ def restart
 
           @worker_pool = nil
         end
+
+        pubsub.shutdown
+        @pubsub = init_pubsub
```

Comment:
Like the worker pool, I think we can just reset to nil here.

.. and we need to do it inside the synchronize.